pymodbus/other_message.py

```diff
@@ -369,7 +369,16 @@ def execute(self, context=None):
             reportSlaveIdData = getattr(context, 'reportSlaveIdData', None)
         if not reportSlaveIdData:
             information = DeviceInformationFactory.get(_MCB)
-            identifier = "-".join(information.values()).encode()
+
+            # Support identity values as bytes data and regular str data
+            id_data = []
+            for v in information.values():
+                if type(v) is bytes:
```

Comment:
Can we just use `isinstance(b, bytes)` instead ? Since we are going to support only 3.x going forward.

Reply:
@dhoomakethu Any chance this will make it to master?